src/WebJobs.Script/Rpc/RpcMessageConversionExtensions.cs

```diff
@@ -0,0 +1,133 @@
+            }
+        }
+
+        public static TypedData ToRpc(this object value)
+        {
+            TypedData typedData = new TypedData();
+
+            if (value == null)
+            {
+                return typedData;
+            }
+
+            if (value is byte[] arr)
+            {
+                typedData.Bytes = ByteString.CopyFrom(arr);
+            }
+            else if (value is JObject jobj)
+            {
+                typedData.Json = jobj.ToString();
+            }
+            else if (value.GetType().IsGenericType && value.GetType().GetGenericTypeDefinition() == typeof(Dictionary<,>))
+            {
+                typedData.Json = JObject.FromObject(value).ToString();
+            }
+            else if (value is HttpRequest request)
+            {
+                var http = new RpcHttp()
+                {
+                    Url = request.Path.ToString(),
+                    Method = request.Method.ToString()
+                };
+                typedData.Http = http;
+
+                foreach (var pair in request.Query)
+                {
+                    http.Query.Add(pair.Key, pair.Value.ToString());
+                }
+
+                foreach (var pair in request.Headers)
+                {
+                    http.Headers.Add(pair.Key.ToLowerInvariant(), pair.Value.ToString());
+                }
+
+                //if (request.Properties.TryGetValue(HttpExtensionConstants.AzureWebJobsHttpRouteDataKey, out IDictionary<string, object> parameters))
+                //{
+                //    foreach (var pair in parameters)
+                //    {
+                //        http.Params.Add(pair.Key, pair.Value.ToString());
+                //    }
+                //}
+
+                if (request.Body != null && request.ContentLength > 0)
+                {
+                    object body = null;
+                    string rawBody = null;
+
+                    switch (request.ContentType)
+                    {
+                        case "application/json":
+                            var jsonReader = new StreamReader(request.Body, Encoding.UTF8);
+                            rawBody = jsonReader.ReadToEnd();
+                            body = JsonConvert.DeserializeObject(rawBody);
+                            break;
+
+                        case "application/octet-stream":
+                            var length = Convert.ToInt32(request.ContentLength);
+                            var bytes = new byte[length];
+                            request.Body.Read(bytes, 0, length);
+                            body = bytes;
+                            break;
+
+                        default:
+                            var reader = new StreamReader(request.Body, Encoding.UTF8);
+                            body = rawBody = reader.ReadToEnd();
+                            break;
+                    }
```

Comment:
Can we remove this entirely and always send as byte? Let the worker decode.  Even better, if we can have Http extension do the conversion, else this feels like duplicated logic.